### ISSUE TITLE
[CI] DateFormatters.testLocales with specific seed fails in 6.x

### ISSUE THREAD

**Issue by dimitris-athanasiou** [elastic/elasticsearch#34907]
This reproduces 100%.

Build failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+periodic/59/console

Failure:

```
DateFormattersTests.testLocales <<< FAILURES!
   > Throwable #1: junit.framework.AssertionFailedError: Expected exception IllegalArgumentException but no exception was thrown
   >    at __randomizedtesting.SeedInfo.seed([59C1B99F0E79EA38:24680EB67D654268]:0)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2693)
   >    at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2678)
   >    at org.elasticsearch.common.time.DateFormattersTests.testLocales(DateFormattersTests.java:134)
   >    at java.lang.Thread.run(Thread.java:748)
```

Reproduce with:

```
./gradlew :server:test -Dtests.seed=59C1B99F0E79EA38 -Dtests.class=org.elasticsearch.common.time.DateFormattersTests -Dtests.method="testLocales" -Dtests.security.manager=true -Dtests.locale=id-ID -Dtests.timricaaulo -Dcompiler.java=11 -Druntime.java=8
```

**Comment by spinscale** [comment]
backporting #34437 will fix this. I am currently running the tests locally...

**Comment by spinscale** [comment]
Fixed by 2df30402563cee6979dba545a73e86569315de3f in 6.x and 6cbd6ede670067c3f6e4a0bf016a97543d2c6e8e in 6.5
